Ports/.port_include.sh

```diff
@@ -77,9 +80,33 @@ run() {
     echo "+ $@"
     (cd "$workdir" && "$@")
 }
-run_replace_in_file(){
+run_replace_in_file() {
     run perl -p -i -e "$1" $2
 }
+install_launcher() {
+    if [ -z "$launcher_name" ] || [ -z "${launcher_category}" ] || [ -z "${launcher_command}" ]; then
+        return
+    fi
+    script_name="${launcher_name,,}"
+    script_name="${script_name// /}"
+    mkdir -p $DESTDIR/usr/local/libexec
+    cat >$DESTDIR/usr/local/libexec/$script_name <<SCRIPT
+#!/bin/sh
+cd \$(dirname -- $launcher_command)
+exec $launcher_command
+SCRIPT
```

Comment:
```suggestion
    cat >$DESTDIR/usr/local/libexec/$script_name <<SCRIPT
#!/bin/sh
set -e
cd -- "\$(dirname -- $(printf %q "$launcher_command"))"
exec $(printf %q "$launcher_command")
SCRIPT
```

Reply:
If you actually want to expand lancher command into `first_word other_words_as_args` (which wouldn't have worked anyway in your version), you can use this:

```suggestion
    cat >$DESTDIR/usr/local/libexec/$script_name <<SCRIPT
#!/bin/sh
set -e
cd -- "\$(dirname -- $(printf %q "${launcher_command%% *}"))"
exec $(for w in $lancher_command; do
    printf '%q\n' $launcher_command
done)
SCRIPT
```
EXTRA NOTE: This code has a small issue if `$lancher_command` is empty; It can easility be fixed \([see here](https://gist.github.com/emanuele6/5f8cf70fb3f5b1ee6d9053c13ec850b5)\), but, since there is a `-z $lancher_command` check above, we can avoid dealing with it.

Reply:
Hm, how about this?

```
    cat >$DESTDIR/usr/local/libexec/$script_name <<SCRIPT
#!/bin/sh
set -e
cd -- "\$(dirname -- \$(which $(printf %q "${launcher_command%% *}")))"
exec $(for w in $launcher_command; do
    printf '%q ' $w
done)
SCRIPT
```

I added `$(which ...)` so I don't have to specify the full path in the `launcher_command`, e.g. `Terminal -e nyancat`. Plus some minor fixes for variable names from your version.

The script that gets created looks like this:

```
#!/bin/sh
set -e
cd -- "$(dirname -- $(which Terminal))"
exec Terminal -e nyancat 
```

Reply:
Looks like `/bin/dirname` and `cd` don't currently understand `--`. So while obviously more correct that way I'm removing that for now.

It also doesn't support `set`, but well, without `set -e` that's not an issue. Heh.

Reply:
That's fine, but it should be:
```bash
    cat >$DESTDIR/usr/local/libexec/$script_name <<SCRIPT
#!/bin/sh
set -e
cd -- "\$(dirname -- "\$(which -- $(printf %q "${launcher_command%% *}"))")"
exec $(for w in $launcher_command; do
    printf '%q ' $w
done)
SCRIPT
```
```
#!/bin/sh
set -e
cd -- "$(dirname -- "$(which -- Terminal)")"
exec Terminal -e nyancat
```

Reply:
I've re-added the `--`s for `dirname` and `which` because apparently only `cd` doesn't support that. I've updated the PR again with the changes. :)

Reply:
@gunnarbeutner: I just noticed that: `printf '%q ' $launcher_command` (with a space after `%q`) can be used instead of the lopp:
```bash
    cat >$DESTDIR/usr/local/libexec/$script_name <<SCRIPT
#!/bin/sh
set -e
cd -- "\$(dirname -- "\$(which -- $(printf %q "${launcher_command%% *}"))")"
exec $(printf '%q ' $launcher_command)
SCRIPT
```

And also solves that issue with `lancher_command` being empty.

Reply:
Updated. :+1: